lib/rspec/core/example_group.rb

```diff
@@ -340,6 +340,28 @@ def self.include_examples(name, *args, &block)
         find_and_eval_shared("examples", name, caller.first, *args, &block)
       end
 
+      # Clear memoized values when adding/removing examples
+      # @private
+      def self.reset_memoized
+        @descendant_filtered_examples = nil
+        @_descendants = nil
+        @parent_groups = nil
+        @declaration_line_numbers = nil
+        @before_context_ivars = nil
```

Comment:
This is why:

```
 1) RSpec::Core::Example#run memory leaks, see GH-321, GH-1921 releases references to the examples / their ivars
     Failure/Error: expect(get_all.call).to eq opts.fetch(:post_gc)
       
       expected: []
            got: ["after_each", "before_all", "before_each", "passing_example"]
       
```

> https://travis-ci.org/rspec/rspec-core/jobs/86674830

I agree it doesn't seem like it needs to be cleared. If I don't clear it though then that test fails.